### DISCUSSION TITLE
Importing Prisma namespace in ESM?

### DISCUSSION THREAD

**Discussion by psibean** [prisma/prisma#11657] [Q&A] [answered]
I have a TypeScript project setup to transpile to an ES2020 module, there are a variety of reasons I have had to move from commonjs. I'm having issues with Prisma and some types.

```
SyntaxError: Named export 'Prisma' not found. The requested module '@prisma/client' is a CommonJS module, which may not support all module.exports as named exports.
CommonJS modules can always be imported via the default export, for example using:

import pkg from '@prisma/client';
const { Prisma } = pkg;
```

But when I do that, I get a namespace error:

```
Cannot find namespace 'Prisma'.ts(2503)
```

Which then prompts me to add the original import

Error is on lines with types, e.g.:

```
Prisma.validator<Prisma.UserArgs>()({
  select: {
    id: true,
    email: true
  }
});
```

**Comment by psibean** [comment, marked as answer]
My bad I'm dumb.

In this case, I had to:

```
import pkg from '@prisma/client";
import type { Prisma as PrismaType} from '@prisma/client'

const { Prisma } = pkg;

// use Prisma.validator and PrismaType.userArgs

```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
